libviso2/package.xml

```diff
@@ -25,6 +25,6 @@
   <buildtool_depend>catkin</buildtool_depend>
 
   <export>
-    <cpp lflags="-L${prefix}/lib -Wl,-rpath,${prefix}/lib -lviso2" cflags="-msse3 -I${prefix}/libviso2/src/"/>
+    <cpp lflags="-L${prefix}/lib -Wl,-rpath,${prefix}/lib -lviso2" cflags="-I${prefix}/libviso2/src/"/>
```

Comment:
Is it possible to make an intel/arm case discrimination here?

Reply:
Is this tested on RPI3?

Reply:
It compiles. I did not do any runtime tests yet

Reply:
home/ubuntu/vo_ws/src/viso2-arm/libviso2/libviso2/src/matcher.cpp: In member function ‘uint8_t* Matcher::createHalfResolutionImage(uint8_t*, const int32_t*)’:
/home/ubuntu/vo_ws/src/viso2-arm/libviso2/libviso2/src/matcher.cpp:639:86: error: ‘_mm_malloc’ was not declared in this scope
   uint8_t* I_half = (uint8_t*)_mm_malloc(dims_half[2]*dims_half[1]*sizeof(uint8_t),16);
                                                                                      ^
/home/ubuntu/vo_ws/src/viso2-arm/libviso2/libviso2/src/matcher.cpp: In member function ‘void Matcher::computeFeatures(uint8_t*, const int32_t*, int32_t*&, int32_t&, int32_t*&, int32_t&, uint8_t*&, uint8_t*&, uint8_t*&, uint8_t*&)’:
/home/ubuntu/vo_ws/src/viso2-arm/libviso2/libviso2/src/matcher.cpp:659:68: error: ‘_mm_malloc’ was not declared in this scope
     I_du = (uint8_t*)_mm_malloc(dims[2]*dims[1]*sizeof(uint8_t*),16);
                                                                    ^
/home/ubuntu/vo_ws/src/viso2-arm/libviso2/libviso2/src/matcher.cpp:669:91: error: ‘_mm_malloc’ was not declared in this scope
     I_du      = (uint8_t*)_mm_malloc(dims_matching[2]*dims_matching[1]*sizeof(uint8_t*),16);
                                                                                           ^
/home/ubuntu/vo_ws/src/viso2-arm/libviso2/libviso2/src/matcher.cpp:679:24: error: ‘_mm_free’ was not declared in this scope
     _mm_free(I_matching);
                        ^
/home/ubuntu/vo_ws/src/viso2-arm/libviso2/libviso2/src/matcher.cpp:698:16: error: ‘_mm_free’ was not declared in this scope
   _mm_free(I_f1);
                ^
/home/ubuntu/vo_ws/src/viso2-arm/libviso2/libviso2/src/matcher.cpp:713:65: error: ‘_mm_malloc’ was not declared in this scope
     max1 = (int32_t*)_mm_malloc(sizeof(Matcher::maximum)*num1,16);
                                                                 ^
/home/ubuntu/vo_ws/src/viso2-arm/libviso2/libviso2/src/matcher.cpp:724:65: error: ‘_mm_malloc’ was not declared in this scope
     max2 = (int32_t*)_mm_malloc(sizeof(Matcher::maximum)*num2,16);
                                                                 ^
[ 10%] Built target std_msgs_generate_messages_lisp
[ 10%] Built target std_msgs_generate_messages_nodejs
/home/ubuntu/vo_ws/src/viso2-arm/libviso2/libviso2/src/matcher.cpp: In member function ‘void Matcher::refinement(std::vector<Matcher::p_match>&, int32_t)’:
/home/ubuntu/vo_ws/src/viso2-arm/libviso2/libviso2/src/matcher.cpp:1500:68: error: ‘_mm_malloc’ was not declared in this scope
   uint8_t* desc_buffer = (uint8_t*)_mm_malloc(32*sizeof(uint8_t),16);
                                                                    ^
/home/ubuntu/vo_ws/src/viso2-arm/libviso2/libviso2/src/matcher.cpp:1583:23: error: ‘_mm_free’ was not declared in this scope
   _mm_free(desc_buffer);
                       ^

Reply:
I am getting the above error

Reply:
Ah, I did not test compiling on Intel yet. Unfortunately I won't have access to my ROS systems until next week. I can investigate further then.

Reply:
Ok, I tested the PR branch\* for successful  builds on:
- RPI3b+ running the newest image from https://downloads.ubiquityrobotics.com/pi.html, ROS Kinetic
    - build command: `catkin_make -j1 --pkg viso2_ros --cmake-args -DARM_CROSS_COMPILATION=ON`
- MacBook Pro (2015), running Ubuntu 16.04., ROS Kinetic
    - build command: `catkin_make -j4 --pkg viso2_ros`

\* Use this git command to clone and checkout this PR's branch: `git clone https://github.com/papr/viso2.git --branch arm`

Reply:
Btw, I had the same error as you did. This is why included this subtle change: https://github.com/miquelmassot/viso2/pull/1/files#diff-59f13c63a111f99623c21195c5c7e3a5R14

Reply:
Tried and tested on RPI3 . Works fine . (:+1: 

git clone https://github.com/papr/viso2.git --branch arm